Demo/src/main/java/com/paypal/android/di/NetworkModule.kt

```diff
@@ -18,27 +23,49 @@ private const val WRITE_TIMEOUT_IN_SEC = 30L
 @InstallIn(SingletonComponent::class)
 object NetworkModule {
 
-    @Provides
-    fun provideRetrofitService(): Retrofit {
+    private fun provideRetrofitService(
+        baseUrl: String,
+        vararg interceptors: Interceptor
+    ): Retrofit {
         return Retrofit.Builder()
-            .baseUrl("https://ppcp-sample-merchant-sand.herokuapp.com")
-            .client(initHttpClientAndInterceptor())
+            .baseUrl(baseUrl)
+            .client(initHttpClientAndInterceptor(*interceptors))
             .addConverterFactory(GsonConverterFactory.create())
             .build()
     }
 
     @Provides
     fun providePaypalApi(): PayPalDemoApi {
-        return provideApi<PayPalDemoApi>(provideRetrofitService())
+        return provideApi(provideRetrofitService("https://ppcp-sample-merchant-sand.herokuapp.com"))
```

Comment:
Let's extract the String URLs - looks like there is some re-use and might be helpful to indicate what each of these URLs is.